src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
-	_(ER_ALTER_SPACE, 12,			"Can't modify space '%s': %s") \
-	_(ER_INDEX_TYPE, 13,			"Unsupported index type supplied for index '%s' in space '%s'") \
-	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index '%s' in space '%s': %s") \
-	_(ER_LAST_DROP, 15,			"Can't drop the primary key in a system space, space '%s'") \
-	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached: %u") \
-	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key in space '%s' while secondary keys exist") \
-	_(ER_KEY_PART_TYPE, 18,			"Supplied key type of part %u does not match index part type: expected %s") \
-	_(ER_EXACT_MATCH, 19,			"Invalid key part count in an exact match (expected %u, got %u)") \
-	_(ER_INVALID_MSGPACK, 20,		"Invalid MsgPack - %s") \
-	_(ER_PROC_RET, 21,			"msgpack.encode: can not encode Lua type '%s'") \
+	_(ER_INJECTION, 8,			"Error injection", "injection_place", STRING) \
+	_(ER_CREATE_SPACE, 9,			"Failed to create space", "space", STRING, "reason", STRING) \
+	_(ER_SPACE_EXISTS, 10,			"Space already exists", "space", STRING) \
+	_(ER_DROP_SPACE, 11,			"Can't drop space", "space", STRING, "reason", STRING) \
+	_(ER_ALTER_SPACE, 12,			"Can't modify space", "space", STRING, "reason", STRING) \
+	_(ER_INDEX_TYPE, 13,			"Unsupported index type", "index", STRING, "space", STRING) \
+	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index", "index", STRING, "space", STRING, "reason", STRING) \
+	_(ER_LAST_DROP, 15,			"Can't drop the primary key in a system space", "space", STRING) \
+	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached", "limit", UINT) \
+	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key while secondary keys exist", "space", STRING) \
+	_(ER_KEY_PART_TYPE, 18,			"Supplied key field type does not match index part type", "fieldno", UINT, "part_type", STRING) \
+	_(ER_EXACT_MATCH, 19,			"Invalid key part count in an exact match", "expected", UINT, "actual", UINT) \
+	_(ER_INVALID_MSGPACK, 20,		"Invalid MsgPack", "issue", STRING) \
+	/* ER_PROC_RET, 21, Unused */ \
 	_(ER_TUPLE_NOT_ARRAY, 22,		"Tuple/Key must be MsgPack array") \
-	_(ER_FIELD_TYPE, 23,			"Tuple field %s type does not match one required by operation: expected %s, got %s") \
-	_(ER_INDEX_PART_TYPE_MISMATCH, 24,	"Field %s has type '%s' in one index, but type '%s' in another") \
-	_(ER_UPDATE_SPLICE, 25,			"SPLICE error on field %s: %s") \
-	_(ER_UPDATE_ARG_TYPE, 26,		"Argument type in operation '%c' on field %s does not match field type: expected %s") \
-	_(ER_FORMAT_MISMATCH_INDEX_PART, 27,	"Field %s has type '%s' in space format, but type '%s' in index definition") \
-	_(ER_UNKNOWN_UPDATE_OP, 28,		"Unknown UPDATE operation #%d: %s") \
-	_(ER_UPDATE_FIELD, 29,			"Field %s UPDATE error: %s") \
+	_(ER_FIELD_TYPE, 23,			"Tuple field type does not match one required by operation", "field", STRING, "expected", STRING, "actual", STRING) \
+	_(ER_INDEX_PART_TYPE_MISMATCH, 24,	"Field type mismatch in indexes", "field", STRING, "type_1", STRING, "type_2", STRING) \
```

Comment:
Probably, better use "actual" and "expected".

And the message should be something like "Index part type does not match space format field type".

Reply:
Looks like the error is about mismatching indexes, at least in `tuple_format_use_key_part()`.

May be `"Field type does not match in indexes", "actual", "expected"`.